tvloader/parser2v1/parse_file_test.go

```diff
@@ -885,4 +886,58 @@ func TestParser2_1FailsIfArtifactURIBeforeArtifactName(t *testing.T) {
 	}
 }
 
+func TestParser2_1FilesWithoutSpdxIdThrowError(t *testing.T) {
+	// case 1
+	// Last unpackaged file no packages in doc
+	// Last file of last package in the doc
+	tvPairs := []reader.TagValuePair{
+		{Tag: "SPDXVersion", Value: "SPDX-2.1"},
+		{Tag: "DataLicense", Value: "CC0-1.0"},
+		{Tag: "SPDXID", Value: "SPDXRef-DOCUMENT"},
+		{Tag: "FileName", Value: "f1"},
+	}
+	_, err := ParseTagValues(tvPairs)
+	if err == nil {
+		t.Errorf("files withoutSpdx Identifiers getting accepted")
+	}
+
+	// case 2 : The previous file (packaged or unpackaged does not contain spdxID)
+	tvPairs = append(tvPairs, reader.TagValuePair{Tag: "FileName", Value: "f2"})
+	_, err = ParseTagValues(tvPairs)
+	if err == nil {
+		t.Errorf("%s", err)
+	}
 
+	// case 3 : Invalid file with snippet
+	// Last unpackaged file before the packges start
+	// Last file of a package and New package starts
+	sid1 := spdx.ElementID("s1")
+	parser := tvParser2_1{
+		doc: &spdx.Document2_1{},
+		st:  psCreationInfo2_1,
+	}
+	fileName := "f2.txt"
+	_ = parser.parsePair2_1("FileName", fileName)
+	_ = parser.parsePair2_1("SnippetSPDXID", string(sid1))
```

Comment:
For these two lines, instead of running `parsePair2_1` and throwing away the result, can you instead include them in the `tvParser2_1{}` setup above? And then only call parsePair2_1 for the call below that we actually want to test here.

See [here](https://github.com/spdx/tools-golang/blob/2d832aee40b755130804757f02fa1d1f0921b8c8/tvloader/parser2v1/parse_file_test.go#L15-L20) for an example of setting up `tvParser2_1{}` with more fields than just doc and st. You likely wouldn't be using exactly what's there, but I'd suggest following that model.

Reply:
Sir these two lines are needed as the logic requires the parser.file to point towards the last parsed file (which is invalid)

Reply:
@swinslow